### PR TITLE
Emacs: neil.el should work with compound commands

### PR DESCRIPTION
**Pull request by agzam**
- [x] This PR corresponds to #245

- [x] I have updated the [CHANGELOG.md](https://github.com/babashka/neil/blob/main/CHANGELOG.md) file with a description of the addressed issue.

- [x] I have considered whether I should add more tests covering the code I've changed.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 See the [New Clojure project quickstart](https://blog.michielborkent.nl/new-clojure-project-quickstart.html) blog post for a gentle introduction into `neil`.
 
 ## 0.3.68
-
+- [#245](https://github.com/babashka/neil/issues/245): neil.el - neil-executable-path now can be set to `clj -M:neil`
```
Comment (borkdude) [inline CHANGELOG.md:8]: 0.3.68 is already release, but you can add this above in a new section called `## Unreleased` :)

**Comment by borkdude** [comment]
Could you maybe explain the fix? Why is there a specific regex about clj/clojure?
Does another command like `bb -Sdeps '...' -m babashka.neil` not work?
In lsp-mode you can define `lsp-clojure-custom-server-command` to be a list of strings, would something similar not work here?

https://github.com/emacs-lsp/lsp-mode/blob/c3be413f8545a26ad0334e1dd2b7e65c448d9f22/clients/lsp-clojure.el#L41

File: CHANGELOG.md (hunk 1 above)
Comment (agzam) [reply CHANGELOG.md:8]: This is elisp portion of it though. No straight version relation, no? I mean you don't cut a new version whenever elisp code changes? Or maybe it's the best to keep them both semvers synced - neil's and elisp package's?

Comment (borkdude) [reply CHANGELOG.md:8]: Note that a Github release also contains all the source files as a zip file and the elisp file at the tag 0.3.68 doesn't contain any changes, so we shouldn't list it under that release.

Comment (agzam) [reply CHANGELOG.md:8]: Alright, I'll add it then under `##Unreleased`. But my question about the package version [here](https://github.com/agzam/neil/blob/neil-executable-path-compound-command/neil.el#L9), should I change that too?

Comment (borkdude) [reply CHANGELOG.md:8]: We don't have to synchronize that per se I guess, but most people will use the master version anyway I presume?

Comment (borkdude) [reply CHANGELOG.md:8]: We can of course bump that version at some point

**Comment by agzam** [comment]
Hmm... good point. So, basically, I'm trying to resolve the executable path, i.e. if it's set to "clj -M:neil", it should become something like "/usr/bin/clj -M:neil". But `(executable-find "clj -M:neil") => nil` - it is not smart enough. So I need to resolve only the first part and then stitch the cmd-line string together.

I suppose I can make it work for any generic command - trying to resolve the first part of it.

File: CHANGELOG.md (hunk 1 above)
Comment (agzam) [reply CHANGELOG.md:9]: I bumped the package version. Will try to keep it aligned with "the last known compatible version" of the tool itself.

**Comment by a13** [comment]
@agzam Just in case, `shell-command-to-string` works even if I use a command name without the full path, it has to be in $PATH though (as well as with `executable-find` case), so the solution looks a bit overcomplicated to me.

**Comment by agzam** [comment]
> the solution looks a bit overcomplicated to me.

executable-find pattern is quite common for this kind of cases, shell-commands may fail if:

- The executable isn't in the current PATH
- The PATH env in Emacs differs from your shell's PATH
- The executable is found but lacks permissions

executable-find helps with:

- avoiding ambiguity with shell aliases
- resolving to the actual binary file, avoiding shell-level abstractions like aliases, functions, or built-ins

This isn't an overcomplication worth simplifying.

**Comment by a13** [comment]
> The executable isn't in the current PATH

in this case, most likely it isn't in `exec-path` either

> The PATH env in Emacs differs from your shell's PATH

`exec-path-from-shell` helps with that _a bit_

> minimizing ambiguity, resolving to the actual binary file, avoiding shell-level abstractions like aliases, functions, or built-ins, symlinked executables, systems with multiple Python/Node/etc. versions

Ok, but if we want to avoid the ambiguity and other shell-related problems, `shell-command` isn't the best solution either, why not `call-process` for example?

> you get the same binary regardless of PATH order or shell configuration

or you don't get it at all :) 

An alternative solution would be to use two defcustoms - one for the command itself and another for (optional) args. Like some other Emacs packages do.

**Comment by agzam** [comment]
Like I said: "I don't think it's an overcomplication worth simplifying.". If you disagree, please make another PR, we can even hold off merging this one until then. If it makes simpler, you can even do it in a nested PR.

You do know how the saying goes: "Talk is cheap, send patches". I already spent a good chunk of my Saturday to fix a problem you pulled out of thin air, if the proposed solution is still not bringing any joy, please do improve it, I have no emotional attachment to this code, nor do I have any privilege to dictate how it should be structured.

**Comment by agzam** [comment]
@a13 upon revisiting my own comment, I just realized how awfully angry it may sound for the reader on the other side. I'm sorry, never was my intent. I just want to emphasize, there's no reluctance from me for any change - if you feel anything can be improved, of course, I will happily review any proposals, I just don't like mincing words when something can be better expressed with code.

**Pull request merged**